### PR TITLE
Add support to node v0.12

### PR DESCRIPTION
**Pull request by kessiler**

### CODE REVIEW

**Comment by dylang** [comment]
Hi @kessiler, thanks for giving this a try!

I've reorganized the code, but in doing so I removed the `./bin/cli.js` file. You may want to bring that file back to make your changes easy to follow.

You may also need to add new directories to the `files` whitelist in the `package.json`. That controls what files and directories will be published.

To test your changes, I feel it would be helpful to include the older versions of node in the travis and appveyor config. It should be pretty clear how to add the additional versions, but I can help too.

**Comment by kessiler** [comment]
Done. Also added node 0.11 :)

**Comment by dylang** [comment]
Nice work! I'm not sure why the build broke in appveyor, some npm install issue. I've restarted the job.

**Comment by dylang** [comment]
I'm still not sure why `appveyor` keeps randomly failing with missing babel dependencies. Any ideas?

**Comment by dylang** [comment]
@kessiler Can you try removing this line from the `appveyor.yml` file in your PR:

```yaml
cache:
  - node_modules -> package.json
```

I'm wondering if it's saving the `node_modules` between node versions and they install different deps.

**Comment by dylang** [comment]
Shoot, a dependency (Grunt) just had a major bump, so this line is going to cause it to always fail:

```yaml
  - node lib\cli.js
```

I need to add an option to not fail when dependencies are out of date.

**Comment by dylang** [comment]
I made some changes that are now in `master` that will hopefully help this PR pass.

**Comment by kessiler** [comment]
awesome @dylang 
It should be able to merge now :+1:

File: package.json [hunk 1]
```diff
@@ -2,13 +2,15 @@
   "name": "npm-check",
   "version": "5.1.3",
   "description": "Check for outdated, incorrect, and unused dependencies.",
-  "main": "lib",
+  "main": "bin",
```
Comment (dylang) [inline package.json]: `bin` should only be used when running from the command line, as it requires `cli`. When using `require('npm-check')` it needs to use `lib/index`.

Comment (kessiler) [inline package.json]: done

**Comment by dylang** [comment]
Great work @kessiler!

**Pull request merged**